ludwig/backend/ray.py

```diff
@@ -244,11 +199,23 @@ def train_fn(
 
         torch.cuda.empty_cache()
 
-        train_results = results, trainer.validation_field, trainer.validation_metric
-
+        # Passing objects containing Torch tensors as metrics is not supported as it will throw an
```

Comment:
You should be able to get around this by putting torch tensors into the object store with `ray.put`, and if necessary converting to a numpy array. I am not sure how the checkpoint API here works. Is it writing to object store, or to disk somehow? If the latter, then how does it handle copies across nodes? I think this merits some additional comments / links to documentation.

Reply:
Good question. What we have here is roughly how Antoni recommended we do this– I can ask for clarification on how `session.report` actually handles the artifact. My guess is that it does copies, but not sure how that would work in a multi-node setting.

Reply:
According to Antoni:
> The checkpoint is put in the object store and then retrieved by the Trainable actor to be reported to Tune. It is also persisted on disk by the Trainable (and synced to cloud, if configured to do so)